src/StrokeEngine.cpp

```diff
@@ -1,551 +1,177 @@
+  this->motor = motor;
         
-#ifdef DEBUG_TALKATIVE
-        Serial.print(" _timeOfStroke: " + String(_timeOfStroke));
-        Serial.print(" | _depth: " + String(_depth));
-        Serial.print(" | _stroke: " + String(_stroke));
-        Serial.println(" | _sensation: " + String(_sensation));
-#endif
-
-        if (_taskStrokingHandle == NULL) {
-            // Create Stroke Task
-            xTaskCreatePinnedToCore(
-                this->_strokingImpl,    // Function that should be called
-                "Stroking",             // Name of the task (for debugging)
-                4096,                   // Stack size (bytes)
-                this,                   // Pass reference to this class instance
-                24,                     // Pretty high task priority
-                &_taskStrokingHandle,   // Task handle
-                1                       // Pin to application core
-            ); 
-        } else {
-            // Resume task, if it already exists
-            vTaskResume(_taskStrokingHandle);
-        }
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Started motion task");
-        Serial.println("Stroke Engine State: " + verboseState[_state]);
-#endif
-
-        return true;
-
-    } else {
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Failed to start motion");
-#endif
-        return false;
-
-    }
-}
-
-void StrokeEngine::stopMotion() {
-    // only valid when 
-    if (_state == PATTERN || _state == SETUPDEPTH) {
-        // Set state
-        _state = READY;
-
-        // Stop servo motor as fast as legally allowed
-        servo->setAcceleration(_maxStepAcceleration);
-        servo->applySpeedAcceleration();
-        servo->stopMove();
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Motion stopped");
-#endif
-
-        // Wait for servo stopped
-        while (servo->isRunning());
-
-        // Send telemetry data
-        if (_callbackTelemetry != NULL) {
-            _callbackTelemetry(float(servo->getCurrentPosition() / _motor->stepsPerMillimeter), 0.0, false);
-        }
-    }
+  // Initialize with default values
+  motionBounds bounds = motor->getBounds();
+
+  this->maxDepth = abs(bounds.end - bounds.start);
+  this->depth = this->maxDepth; 
+  this->stroke = this->maxDepth / 3;
+  this->strokeRate = 1.0;
+  this->sensation = 0.0;
+
+  ESP_LOGD("StrokeEngine", "Stroke Parameter Max Depth = %f", this->maxDepth);
+  ESP_LOGD("StrokeEngine", "Stroke Parameter Depth = %f", this->depth);
+  ESP_LOGD("StrokeEngine", "Stroke Parameter Stroke = %f", this->stroke);
+  ESP_LOGD("StrokeEngine", "Stroke Parameter Stroke Rate = %f", this->strokeRate);
+  ESP_LOGD("StrokeEngine", "Stroke Parameter Sensation = %f", this->sensation);
+
+  ESP_LOGI("StrokeEngine", "Attached Motor succesfully to Stroke Engine!");
+}
+
+void StrokeEngine::setParameter(StrokeParameter parameter, float value, bool applyNow) {
+  if (xSemaphoreTake(_patternMutex, portMAX_DELAY) != pdTRUE) {

```

Comment:
Maybe a more generic name for_patternMutex is appropriated. e.g. _parameterMutex